### PR TITLE
Stop stale errors from popping up causing spurious failures when reusing a kept-alive HTTPS connection

### PR DESCRIPTION
**Pull request by pmq20**
## Problem

Stale OpenSSL errors are thrown causing spurious failures when reusing a kept-alive HTTPS connection. A test has been added to reproduce this bug.

* bad_sign() would inject an error into OpenSSL's error stack
* Second time of calling requestThunk would make the error be thrown

## Solution

As a workaround, forcibly clear OpenSSL's error stack. This workaround could be removed when / if the upstream Node.js / io.js project has fixed this bug.

### CODE REVIEW

**Comment by coolme200** [comment]
这个可以先不合，等看看能否在 iojs 层面处理。

File: lib/urllib.js [hunk 1]
```diff
@@ -185,6 +197,7 @@ exports.requestWithCallback = function (url, args, callback) {
   var agent = args.agent || exports.agent;
 
   if (parsedUrl.protocol === 'https:') {
+    workaround_clear_openssl_errors();
```
Comment (fengmk2) [inline lib/urllib.js]: I will add a optional to detect clean openssl errors.

**Comment by pmq20** [comment]
iojs 层面处理尝试：

https://github.com/nodejs/io.js/pull/2342

**Comment by pmq20** [comment]
iojs 层面上的补丁落地了

**Comment by dead-horse** [comment]
:+1:

**Comment by fengmk2** [comment]
这个就不合并了。

**Pull request closed without merging**